test/integration/backward_compatible/parallel/sql/DataTypeTest.js

```diff
@@ -814,4 +815,59 @@ describe('Data type test', function () {
             rows[i]['__key'].should.be.eq(expectedKeys[i]);
         }
     });
+    it('should be able to decode/serialize JSON', async function () {
+        await basicSetup(this);
+        const SqlColumnType = TestUtil.getSqlColumnType();
+
+        // JSON support is added in 5.1.
+        TestUtil.markClientVersionAtLeast(this, '5.1');
+        TestUtil.markServerVersionAtLeast(this, client, '5.1');
+
+        const createMappingQuery = `
+            CREATE MAPPING ${mapName} (
+                __key INT,
+                this JSON
+            )
+            TYPE IMAP
+            OPTIONS (
+                'keyFormat' = 'int',
+                'valueFormat' = 'json'
+            )
+        `;
+
+        await client.getSql().execute(createMappingQuery);
+
+        await client.getSql().execute(`INSERT INTO ${mapName} VALUES (2, CAST('{"age": 2}' AS JSON)),` +
+            '(3, ?)',
+        [new HazelcastJsonValue(JSON.stringify({age: 3}))]);
+
+        const result = await TestUtil.getSql(client).execute(
+            `SELECT * FROM ${mapName} WHERE CAST(JSON_VALUE(this, '$.age') AS DOUBLE) > ?` +
+            ' AND CAST(JSON_VALUE(this, \'$.age\') AS DOUBLE) < ? ORDER BY __key ASC',
```

Comment:
Is string concatenation required? Can't you just put the second part of the string in the backtick string above?
```
`SELECT * ...
 AND CAST ...
`
```

Reply:
it will include the spaces and newline characters after the `select * ...` part. It won't work most probably

Reply:
Ok changed that part